src/ui/component/field/field.tsx

```diff
@@ -1,21 +1,23 @@
-import { type FC, type ChangeEvent, useCallback, useState, useRef } from 'react'
+import { type FC, type ChangeEvent, useCallback, useState, useRef, type ReactNode } from 'react'
 import classNames from 'classnames'
 import { Icon } from '@/ui/component/icon/icon'
 import './field.scss'
 
 type FieldProps = {
   id: string
-  label: string
+  label?: string
   onChange: (value: string) => void
   value?: string
   type?: 'text' | 'number'
   error?: string
+  placeholder?: string
   required?: boolean
   disabled?: boolean
+  readonly?: boolean
   multiline?: boolean
   resizable?: boolean
   leftElement?: JSX.Element
-  rightElement?: JSX.Element
+  rightElement?: ReactNode
```

Comment:
Just for my comprehension, why `rightElement` becomes a `ReactNode`?
Be careful with `ReactNode` because it accepts everything: 

`type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;`

By the way, why `leftElement` remains a `JSX.Element` ?

Reply:
I'm not sure about the all possible `leftElement` options, so didn't want to change that yet. For the `rightElement` we know already that there is a case where we need to add `string`, such as 'KNOW' in the step 3 and 4. We tried wrapping that text in <span> tag initially, but it was not matching the style we needed so we thought of replacing JSX.Element to ReactNode. Would you say this is not ideal solution, should I try something else?

Reply:
Is `JSX.Element | string` a more suitable alternative for limiting the `rightElement` type to what we need?

Reply:
By the way, if you need a string, you can keep a single JSX.Element by doing this instead:

```tsx
rightELement={<span>KNOW</span>}
```

Reply:
Fixed now!